Assets/Projectile.cs

```diff
@@ -28,12 +30,13 @@ void FixedUpdate()
     void OnTriggerEnter2D(Collider2D other)
     {
         Health health = other.gameObject.GetComponent<Health>();
+        Team otherTeam = other.GetComponent<Team>();
 
-        // check whether the colliding object can be damaged
-        if (health)
+        // check whether the colliding object can be damaged and can not harm objects that are on the same team
+        if (health && otherTeam.GetTeam() != this.team.GetTeam())
```

Comment:
Neutral enemies are enemies with everyone. Objects with no team (or team == none) are truly neutral / friendly with everyone (note that either case will fail to pass the boolean check for team &&....). Perhaps this is bad because none is only treated as "friendly" for both sides by not having a team object. If we were to test it, assigning none would return true for IsEnemy and false for IsFriendly.

Either way in this case I would argue that we should use IsEnemy and IsFriendly and handle "how should we treat such and such an enemy" within changing or adding functionality to the Team class.

The team check should be there just to confirm the Team class was found on the object before calling a function from it. This will properly handle when team is null or set to none. If we want to improve that, we should fix IsEnemy and IsFriendly to return the proper value for case team == none as well.